File: 5-Problem-A-Day/67. Add Binary.java
class Solution {
    public String addBinary(String a, String b) {
        int i = a.length() - 1, j = b.length() - 1;
        int carry = 0;
        StringBuilder ans = new StringBuilder();
        while (i > -1 || j > -1) {
            int tmp = carry;
            if (i > -1)
                tmp += Character.getNumericValue(a.charAt(i));
            if (j > -1)
                tmp += Character.getNumericValue(b.charAt(j));
            if (tmp <= 1) {
                ans.append(Integer.toString(tmp));
                carry = 0;
            } else {
                ans.append(Integer.toString(tmp % 2));
                carry = 1;
            }n
            i -= 1;
            j -= 1;
        }
        if (carry > 0)
            ans.append(Integer.toString(carry));
        ans.reverse();
        return ans.toString();
    }
}

File: 2021/67. Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        string ans;
        int i = a.length() - 1, j = b.length() - 1;
        int carry = 0;
        while (i >= 0 || j >= 0 || carry > 0) {
            int tmp = carry;
            if (i >= 0) {
                tmp += a[i] - '0';
                i--;
            } 
            if (j >= 0) {
                tmp += b[j] - '0';
                j--;
            }
            ans += (tmp % 2) + '0';
            carry = tmp / 2;
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};